src/frontend/components/Chat/SharedChatComponents/ChatConversationDisplayer/index.spec.tsx

```diff
@@ -0,0 +1,89 @@
+import { act, render } from '@testing-library/react';
+import React from 'react';
+import { ChatConversationDisplayer } from '.';
+import { wrapInIntlProvider } from '../../../../utils/tests/intl';
+import { useMessagesState } from 'data/stores/useMessagesStore';
+import { DateTime } from 'luxon';
+
+const mockScrollIntoView = jest.fn();
+window.HTMLElement.prototype.scrollIntoView = mockScrollIntoView;
+
+describe('<ChatConversationDisplayer />', () => {
+  beforeEach(() => {
+    jest.useRealTimers();
+    mockScrollIntoView.mockReset();
+  });
+
+  it('checks spinner is displayed first, and then chat messages', async () => {
+    jest.useFakeTimers();
+    const { container } = render(
+      wrapInIntlProvider(<ChatConversationDisplayer />),
+    );
+
+    expect(container).toMatchSnapshot();
+    await act(async () => {
+      jest.advanceTimersByTime(4000);
+    });
+
+    act(() =>
+      useMessagesState.getState().addMessage({
+        dateTime: DateTime.fromISO('2020-01-01T12:12:12'),
+        sender: 'John Doe',
+        content: 'This is an exemple message.',
+      }),
+    );
+    const messageTimeSpan = container.getElementsByClassName('fiMORS')[0];
+    const senderNameSpan = container.getElementsByClassName('iJaXLa')[0];
+    const messageContentSpan = container.getElementsByClassName('bTtMyZ')[0];
+    expect(messageTimeSpan).toHaveTextContent('12:12');
```

Comment:
Why not doing `screen.getByText('12:12');` ?

Reply:
done in [b39cfdc22c60ea5a6d0fded73af5d64aebaaee9d](https://github.com/openfun/marsha/pull/1247/commits/b39cfdc22c60ea5a6d0fded73af5d64aebaaee9d)